src/url.php

```diff
@@ -320,6 +321,36 @@ function qtranxf_get_url_info( string $url ): array {
     return $urlinfo;
 }
 
+/**
+ * Returns the base admin url of the WordPress backend name e.g. wp-admin
+ * If the $admin_base is empty it returns the standard WordPress backend name
+ *
+ * @author Sebastian Poetter https://github.com/poetter-sebastian
+ * @link https://github.com/qtranslate/qtranslate-xt/pull/1324 repo pull request
+ *
+ * @return string WordPress backend name
+ */
+function qtranxf_get_admin_base():string
+{
+    $admin_base = str_replace( [site_url(), '/'], '', admin_url() );
+    return strlen($admin_base) == 0? 'wp-admin' : $admin_base;
+}
+
+/**
+ * Returns the base admin url of the WordPress backend login url e.g. wp-login
+ * If the $login_base is empty it returns the standard WordPress backend login name
+ *
+ * @author Sebastian Poetter https://github.com/poetter-sebastian
+ * @link https://github.com/qtranslate/qtranslate-xt/pull/1324 repo pull request
+ *
+ * @return string WordPress backend login name
+ */
+function qtranxf_get_login_base():string
+{
+    $login_base = str_replace( [site_url(), '/', '.php'], '', wp_login_url() );
+    return strlen($login_base) == 0? 'wp-login' : $login_base;
```

Comment:
- Same as above, just use and return `trim`.
- There is something wrong with the original regex I think, as the `wp-login.php` case is already covered by the first block (`wp-.*\.php`). Anyways if you want to extend to the case of custom bases, there is no need to strip the `'.php'`, as what we want to get is actually `wp-login.php` rather than `wp-login` only. And `'.php'` could be part of the custom base also.